python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -0,0 +1,342 @@
+            self.message_types = set([(t.get_type() if MessagePayload.is_subclass(t) else t) for t in message_types])
+
+        self.valid_count = 0
+        self.message_counts = {}
+        self.prev_sequence_number = None
+        self.total_bytes_read = 0
+
+        self.show_progress = show_progress
+        self.last_print_bytes = 0
+        self.start_time = datetime.now()
+
+        # Open the file to be read.
+        if isinstance(input_file, str):
+            self.input_file = open(input_file, 'rb')
+        else:
+            self.input_file = input_file
+
+        input_path = self.input_file.name
+        self.file_size_bytes = os.stat(input_path).st_size
+
+        # Open the companion index file if one exists.
+        self.index_path = file_index.FileIndex.get_path(input_path)
+        self.next_index_elem = 0
+        if ignore_index:
+            if os.path.exists(self.index_path):
+                self.logger.debug("Ignoring index file @ '%s'." % self.index_path)
+            self.index = None
+            self.index_builder = file_index.FileIndexBuilder() if generate_index else None
+        else:
+            if os.path.exists(self.index_path):
+                try:
+                    self.logger.debug("Loading index file '%s'." % self.index_path)
+                    self.index = file_index.FileIndex(index_path=self.index_path, data_path=input_path,
+                                                      delete_on_error=generate_index)
+                    self.index = self.index[self.message_types][self.time_range]
+                    self.index_builder = None
+                except ValueError as e:
+                    self.logger.error("Error loading index file: %s" % str(e))
+                    self.index = None
+                    self.index_builder = file_index.FileIndexBuilder() if generate_index else None
+            else:
+                self.logger.debug("No index file found @ '%s'." % self.index_path)
+                self.index = None
+                self.index_builder = file_index.FileIndexBuilder() if generate_index else None
+
+        if self.index_builder is not None:
+            self.logger.debug("Generating index file '%s'." % self.index_path)
+
+    def have_index(self):
+        return self.index is not None
+
+    def get_index(self):
+        return self.index
+
+    def generating_index(self):
+        return self.index_builder is not None
+
+    def get_bytes_read(self):
+        return self.total_bytes_read
+
+    def next(self):
+        while True:
+            if not self._advance_to_next_sync():
+                # End of file.
+                self.total_bytes_read = self.input_file.tell()
+                self.logger.debug('EOF reached.')
+                break
+
+            offset_bytes = self.input_file.tell()
+            self.total_bytes_read = offset_bytes
+            self._print_progress()
+
+            if self.total_bytes_read - self.last_print_bytes > 10e6:
```

Comment:
Isn't this just the repeated contents of `_print_progress()`? Did you mean to remove this?

Reply:
Yep meant to delete this stuff.